File: clients/Go/5.0 mcp_store_user/MCPStoreUserClient.go
package main

import (
	"bytes"
	"encoding/json"
	"errors"
	"flag"
	"fmt"
	"io"
	"net"
	"os"
)

type StoreUserPayload struct {
	Command   string                 `json:"command"`
	Token     string                 `json:"token"`
	Arguments map[string]interface{} `json:"arguments"`
}

func sendStoreUserRequest(serverIP string, serverPort int, token, name, email, password, language, timezone string, roles, groups []string, usePublic, activateFtp bool, ftpPassword string) (string, error) {
	// Prepare the request payload
	payload := StoreUserPayload{
		Command: "store_user",
		Token:   token,
		Arguments: map[string]interface{}{
			"name":        name,
			"email":       email,
			"password":    password,
			"language":    language,
			"timezone":    timezone,
			"roles":       roles,
			"groups":      groups,
			"usePublic":   usePublic,
			"activateFtp": activateFtp,
			"ftpPassword": ftpPassword,
		},
	}

	// Convert the payload to JSON
	payloadJSON, err := json.Marshal(payload)
	if err != nil {
		return "", err
	}

	// Create a connection to the server
	conn, err := net.Dial("tcp", fmt.Sprintf("%s:%d", serverIP, serverPort))
	if err != nil {
		return "", err
	}
	defer conn.Close()

	// Send the request
	_, err = conn.Write(payloadJSON)
	if err != nil {
		return "", err
	}

	// Receive the response
	var responseBuffer bytes.Buffer
	buf := make([]byte, 4096)
	for {
		n, err := conn.Read(buf)
		if err != nil {
			if errors.Is(err, io.EOF) {
				break
			}
			return "", err
		}
		responseBuffer.Write(buf[:n])
		if n < 4096 {
			break
		}
	}

	return responseBuffer.String(), nil
}

func main() {
	serverIP := flag.String("server-ip", "", "IP address of the MCP server")
	serverPort := flag.Int("server-port", 0, "Port number of the MCP server")
	token := flag.String("token", "", "Authentication token")
	name := flag.String("name", "", "Name of the user")
	email := flag.String("email", "", "Email of the user")
	password := flag.String("password", "", "Password for the user")
	language := flag.String("language", "en", "Language code (optional)")
	timezone := flag.String("timezone", "Europe/Berlin", "Timezone (optional)")
	roles := flag.String("roles", "", "Comma-separated list of roles for the user (optional)")
	groups := flag.String("groups", "", "Comma-separated list of groups for the user (optional)")
	usePublic := flag.Bool("usePublic", false, "Use the public knowledge base")
	activateFtp := flag.Bool("activateFtp", false, "Activate FTP for the user")
	ftpPassword := flag.String("ftpPassword", "", "FTP password for the user (optional)")

	flag.Parse()

	if *serverIP == "" || *serverPort == 0 || *token == "" || *name == "" || *email == "" || *password == "" {
		fmt.Println("❌ ERROR: Required flags are missing.")
		flag.Usage()
		os.Exit(1)
	}

	// Convert roles and groups to slices
	roleList := []string{}
	if *roles != "" {
		roleList = append(roleList, *roles)
	}

groupList := []string{}
	if *groups != "" {
		groupList = append(groupList, *groups)
	}

	// Send the request to store the user
	fmt.Println("📤 Sending request to create a new user...")
	response, err := sendStoreUserRequest(*serverIP, *serverPort, *token, *name, *email, *password, *language, *timezone, roleList, groupList, *usePublic, *activateFtp, *ftpPassword)
	if err != nil {
		fmt.Printf("❌ ERROR: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("✔️ Response from server:")
	fmt.Println(response)
}

File: clients/Python/5.0 mcp_store_user/MCPStoreUserClient.py
import socket
import ssl
import json
import argparse
import sys

def send_store_user_request(server_ip, server_port, token, name, email, password, language, timezone, roles, groups, usePublic, activateFtp, ftpPassword, use_ssl=True, accept_self_signed=False):
    """
    Sends a request to create a new user on the MCP server.
    """
    payload = {
        "command": "store_user",
        "token": token,
        "arguments": {
            "name": name,
            "email": email,
            "password": password,
            "language": language,
            "timezone": timezone,
            "roles": roles,
            "groups": groups,
            "usePublic": usePublic,
            "activateFtp": activateFtp,
            "ftpPassword": ftpPassword
        }
    }

    payload_json = json.dumps(payload)
    
    raw_socket = None
    client_socket = None

    try:
        raw_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        raw_socket.settimeout(10)

        if use_ssl:
            context = ssl.create_default_context()
            if accept_self_signed:
                context.check_hostname = False
                context.verify_mode = ssl.CERT_NONE
            client_socket = context.wrap_socket(raw_socket, server_hostname=server_ip)
        else:
            client_socket = raw_socket
        
        client_socket.connect((server_ip, server_port))
        client_socket.sendall(payload_json.encode('utf-8'))

        response = b""
        while True:
            part = client_socket.recv(4096)
            if not part:
                break
            response += part

        return response.decode('utf-8')
    except ssl.SSLError:
        return "Error: Server and/or client may require TLS encryption. Please enable SSL/TLS."
    except Exception as e:
        return f"Error: {e}"
    
    finally:
        if client_socket is not None:
            try:
                client_socket.shutdown(socket.SHUT_RDWR)
            except:
                pass
            client_socket.close()

if __name__ == "__main__":
    parser = argparse.ArgumentParser(
        description="Send a request to create a new user on the MCP server.",
        formatter_class=argparse.RawTextHelpFormatter
    )
    parser.add_argument("--server-ip", required=True, help="IP address of the MCP server")
    parser.add_argument("--server-port", required=True, type=int, help="Port number of the MCP server")
    parser.add_argument("--token", required=True, help="Authentication token")
    parser.add_argument("--name", required=True, help="Name of the user")
    parser.add_argument("--email", required=True, help="Email of the user")
    parser.add_argument("--password", required=True, help="Password for the user")
    parser.add_argument("--language", default="en", help="Language code (optional)")
    parser.add_argument("--timezone", default="Europe/Berlin", help="Timezone (optional)")
    parser.add_argument("--roles", nargs="+", default=[], help="Roles for the user (optional)")
    parser.add_argument("--groups", nargs="+", default=[], help="Groups for the user (optional)")
    parser.add_argument("--usePublic", action="store_true", help="Use the public knowledge base")
    parser.add_argument("--activateFtp", action="store_true", help="Activate FTP for the user")
    parser.add_argument("--ftpPassword", default="", help="FTP password for the user (optional)")
    parser.add_argument("--use-ssl", action="store_true", help="Connect using SSL/TLS")
    parser.add_argument("--accept-self-signed", action="store_true", help="Accept self-signed certificates (disable certificate verification)")

    if len(sys.argv) == 1:
        parser.print_help(sys.stderr)
        sys.exit(1)

    args = parser.parse_args()

    response = send_store_user_request(
        args.server_ip,
        args.server_port,
        args.token,
        args.name,
        args.email,
        args.password,
        args.language,
        args.timezone,
        args.roles,
        args.groups,
        args.usePublic,
        args.activateFtp,
        args.ftpPassword,
        use_ssl=args.use_ssl,
        accept_self_signed=args.accept_self_signed
    )
    print("Response from server:", response)
